crates/pico/src/memo.rs

```diff
@@ -156,16 +162,17 @@ fn derived_node_changed_since(db: &Database, derived_node_id: DerivedNodeId, sin
         return true;
     };
     let did_recalculate = memo(db, derived_node_id, inner_fn);
-    matches!(did_recalculate, DidRecalculate::Recalculated)
+    matches!(
+        did_recalculate,
+        DidRecalculate::Recalculated | DidRecalculate::Error
```

Comment:
I think what's happening is that if any nested memoized function fails with `DidCalculate::Error`, then that propagates all the way up to whatever top-level function we have.

So that's good